workitem/workitem_repository_blackbox_test.go

```diff
@@ -347,3 +405,270 @@ func (s *workItemRepoBlackBoxTest) TestConcurrentWorkItemCreations() {
 		assert.Equal(s.T(), 0, report.failures)
 	}
 }
+
+// given a bunch of tests with expected error results for each work item
+// type field kind, a work item type for each kind...
+type validInvalid struct {
+	Valid   []interface{}
+	Invalid []interface{}
+	// When the actual value is a zero (0), it will be interpreted as a
+	// float64 rather than an int. To compensate for that ambiguity, a
+	// kind can opt-in to provide an construction function that returns
+	// the correct value.
+	Compensate func(interface{}) interface{}
+}
+
+// getFieldTypeTestData returns a list of legal and illegal values to be used
+// with a given field type (here: the map key).
+func getFieldTypeTestData(t *testing.T) map[workitem.Kind]validInvalid {
+	// helper function to convert a string into a duration and handling the
+	// error
+	validDuration := func(s string) time.Duration {
+		d, err := time.ParseDuration(s)
+		if err != nil {
+			require.Nil(t, err, "we expected the duration to be valid: %s", s)
+		}
+		return d
+	}
+
+	return map[workitem.Kind]validInvalid{
+		workitem.KindString: {
+			Valid: []interface{}{
+				"foo",
+			},
+			Invalid: []interface{}{
+				"", // NOTE: an empty string is not allowed in a required field.
+				nil,
+				0,
+				true,
+				0.1,
+			},
+		},
+		workitem.KindUser: {
+			Valid: []interface{}{
+				"jane doe", // TODO(kwk): do we really allow usernames with spaces?
+				"",         // TODO(kwk): do we really allow empty usernames?
+			},
+			Invalid: []interface{}{
+				nil,
+				0,
+				true,
+				0.1,
+			},
+		},
+		workitem.KindIteration: {
+			Valid: []interface{}{
+				"some iteration name",
+				"", // TODO(kwk): do we really allow empty iteration names?
+			},
+			Invalid: []interface{}{
+				nil,
+				0,
+				true,
+				0.1,
+			},
+		},
+		workitem.KindArea: {
+			Valid: []interface{}{
+				"some are name",
+				"", // TODO(kwk): do we really allow empty area names?
+			},
+			Invalid: []interface{}{
+				nil,
+				0,
+				true,
+				0.1,
+			},
+		},
+		workitem.KindLabel: {
+			Valid: []interface{}{
+				"some label name",
+				"", // TODO(kwk): do we really allow empty label names?
+			},
+			Invalid: []interface{}{
+				nil,
+				0,
+				true,
+				0.1,
+			},
+		},
+		workitem.KindURL: {
+			Valid: []interface{}{
+				"127.0.0.1",
+				"http://www.openshift.io",
+				"openshift.io",
```

Comment:
you could also add `http://localhost:8080/foo` to verify the presence of a port number and a path

Reply:
that should already be covered by the line below which goes like this: 

```go
"ftp://url.with.port.and.different.protocol.port.and.parameters.com:8080/fooo?arg=bar&key=value"
```